contrib/epee/include/net/levin_protocol_handler_async.h

```diff
@@ -372,7 +359,7 @@ class async_protocol_handler
 
     // Never call callback inside critical section, that can cause deadlock. Callback can be called when
     // invoke_response_handler_base is cancelled
-    std::for_each(local_invoke_response_handlers.begin(), local_invoke_response_handlers.end(), [](const boost::shared_ptr<invoke_response_handler_base>& pinv_resp_hndlr) {
+    std::for_each(local_invoke_response_handlers.begin(), local_invoke_response_handlers.end(), [](const std::shared_ptr<invoke_response_handler_base>& pinv_resp_hndlr) {
       pinv_resp_hndlr->cancel();
     });
```

Comment:
```suggestion
    for (const auto& pinv_resp_hndlr : local_invoke_response_handlers) {
      pinv_resp_hndlr->cancel();
    }
```

This is clearer IMO and removes the possibility of unnecessary overhead from `std::for_each` and the closure.

Reply:
Your right, since I'm touching this anyway, might as well update to it to C++11+ foreach loops.